src/ServiceControl.Transports.RabbitMQ/QueueLengthProvider.cs

```diff
@@ -0,0 +1,142 @@
+﻿namespace ServiceControl.Transports.RabbitMQ
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using global::RabbitMQ.Client;
+    using Monitoring;
+    using Monitoring.Infrastructure;
+    using Monitoring.Messaging;
+    using Monitoring.QueueLength;
+    using NServiceBus.Logging;
+    using NServiceBus.Metrics;
+
+    public class QueueLengthProvider : IProvideQueueLength
+    {
+        public void Initialize(string connectionString, QueueLengthStore store)
+        {
+            this.connectionString = connectionString;
+            this.store = store;
+        }
+
+        public void Process(EndpointInstanceId endpointInstanceId, EndpointMetadataReport metadataReport)
+        {
+            var endpointInstanceQueue = new EndpointInputQueue(endpointInstanceId.EndpointName, metadataReport.LocalAddress);
+            var queueName = metadataReport.LocalAddress;
+
+            endpointQueues.AddOrUpdate(endpointInstanceQueue, _ => queueName, (_, currentValue) =>
+            {
+                if (currentValue != queueName) sizes.TryRemove(currentValue, out var _);
+
+                return queueName;
+            });
+        }
+
+        public void Process(EndpointInstanceId endpointInstanceId, TaggedLongValueOccurrence metricsReport)
+        {
+            //RabbitMQ does not support endpoint level queue length reports
+        }
+
+        public Task Start()
+        {
+            var factory = new ConnectionFactory
+            {
+                Uri = connectionString,
+                AutomaticRecoveryEnabled = true,
+                TopologyRecoveryEnabled = true
```

Comment:
This one is true by default as well, but I'd recommend setting it to false instead. You don't actually have anything here to recover, so there's no reason to have it enabled.